lib/ecto/tenant/supervisor.ex

```diff
@@ -0,0 +1,64 @@
+defmodule Ecto.Tenant.Supervisor do
+  @moduledoc false
+
+  defmacro __using__(opts \\ []) do
+
+    quote bind_quoted: [opts: opts] do
+
+      defmodule Supervisor do
+
+        @repo opts[:repo]
+
+        def child_spec(config \\ []) do
+          %{
+            id: {__MODULE__, @repo},
+            start: {__MODULE__, :start_link, [config]},
+            type: :supervisor
+          }
+        end
+
+        def start_link(config_arg \\ []) do
+          repo_config = @repo.config()
+          |> Keyword.drop([:repos, :tenants])
+
+          children = case @repo.dynamic_repo_configs() do
+            [] -> [single_repo_child(repo_config, config_arg)]
```

Comment:
This looks like it defaults to using the repo module as the child. Does this work or does it just serve to give the supervisor at least one child?